apps/foundation-app/src/app/design/molecules/qty-dropdown/qty-dropdown.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, OnInit, Input, Renderer2, ElementRef, ViewChild } from '@angular/core';
+
+@Component({
+  selector: 'qty-dropdown',
+  templateUrl: './qty-dropdown.component.html',
+  styleUrls: ['./qty-dropdown.component.scss']
+})
+export class QtyDropdownComponent implements OnInit {
+
+  @Input() qty: number;
+  @Input() id: number;
+
+  static readonly dropdownRange = 9;
+  dropdownItemCounter: number[];
+  inputHasBeenShown: boolean;
```

Comment:
I added this boolean, because when the `<input>` is edited, the qty element would be immediately changed back into a `<select>` when the user would type the tens digit of anything over ten.

Reply:
Ah, that's an interesting problem.... maybe we can do a brief timeout before the check happens with `rxjs` that way we give the user some time before it switches, I think 300ms would be sufficient?

Reply:
I was thinking about doing that, but then I thought of myself as a user going: "Well, we want at least ten (types in "1"), but how many...WTF!" So that's why I ultimately went with this. What do you think?

Reply:
I think this solution works for now.